core/types/wo.go

```diff
@@ -0,0 +1,960 @@
+	if nodeCtx == common.ZONE_CTX {
+		wo.woHeader.number = number
+	} else {
+		wo.woBody.header.SetNumber(number, nodeCtx)
+	}
+}
+
+func (wo *WorkObject) SetDifficulty(difficulty *big.Int) {
+	wo.woHeader.difficulty = difficulty
+}
+
+func (wo *WorkObject) SetNonce(nonce BlockNonce) {
+	wo.woHeader.nonce = nonce
+}
+
+func (wo *WorkObject) SetHeaderHash(headerHash common.Hash) {
+	wo.woHeader.headerHash = headerHash
+}
+
+func (wo *WorkObject) SetLocation(location common.Location) {
+	wo.woHeader.SetLocation(location)
+}
+
+func (wo *WorkObject) SetEVMRoot(root common.Hash) {
+	wo.woBody.header.SetEVMRoot(root)
+}
+
+func (wo *WorkObject) SetParentEntropy(entropy *big.Int, nodeCtx int) {
+	wo.woBody.header.SetParentEntropy(entropy, nodeCtx)
+}
+
+func (wo *WorkObject) SetEtxRollupHash(hash common.Hash) {
+	wo.woBody.header.SetEtxRollupHash(hash)
+}
+
+func (wo *WorkObject) SetBaseFee(fee *big.Int) {
+	wo.woBody.header.SetBaseFee(fee)
+}
+
+func (wo *WorkObject) SetGasUsed(gasUsed uint64) {
+	wo.woBody.header.SetGasUsed(gasUsed)
+}
+
+func (wo *WorkObject) SetGasLimit(gasLimit uint64) {
+	wo.woBody.header.SetGasLimit(gasLimit)
+}
+
+func (wo *WorkObject) SetTime(time uint64) {
+	wo.woBody.header.SetTime(time)
+}
+
+func (wo *WorkObject) SetCoinbase(coinbase common.Address) {
+	wo.woBody.header.SetCoinbase(coinbase)
+}
+
+func (wo *WorkObject) SetManifestHash(hash common.Hash, nodeCtx int) {
+	wo.woBody.header.SetManifestHash(hash, nodeCtx)
+}
+
+func (wo *WorkObject) SetParentDeltaS(deltaS *big.Int, nodeCtx int) {
+	wo.woBody.header.SetParentDeltaS(deltaS, nodeCtx)
+}
+
+func (wo *WorkObject) SetUncleHash(hash common.Hash) {
+	wo.woBody.header.SetUncleHash(hash)
+}
+
+func (wo *WorkObject) SetEtxHash(hash common.Hash) {
+	wo.woBody.header.SetEtxHash(hash)
+}
+
+func (wo *WorkObject) SetReceiptHash(hash common.Hash) {
+	wo.woBody.header.SetReceiptHash(hash)
+}
+
+func (wo *WorkObject) SetExtra(extra []byte) {
+	wo.woBody.header.SetExtra(extra)
+}
+
+func (wo *WorkObject) SetTransactionsRoot(root common.Hash) {
+	wo.woBody.header.SetTxHash(root)
+}
+
+func (wo *WorkObject) SetUTXORoot(root common.Hash) {
+	wo.woBody.header.SetUTXORoot(root)
+}
+
+// GetAppendTime returns the appendTime of the block
+// The appendTime is computed on the first call and cached thereafter.
+func (wo *WorkObject) GetAppendTime() time.Duration {
+	if appendTime := wo.appendTime.Load(); appendTime != nil {
+		if val, ok := appendTime.(time.Duration); ok {
+			return val
+		}
+	}
+	return -1
+}
+
+func (wo *WorkObject) SetAppendTime(appendTime time.Duration) {
+	wo.appendTime.Store(appendTime)
+}
+
+func (wo *WorkObject) QiTransactions() []*Transaction {
+	return wo.woBody.QiTransactions()
+}
+
+func (wo *WorkObject) QuaiTransactions() []*Transaction {
+	return wo.woBody.QuaiTransactions()
+}
+
+func (wo *WorkObject) EncodeRLP(w io.Writer) error {
+	return wo.woBody.header.EncodeRLP(w)
+}
+
+func (wo *WorkObject) Size() common.StorageSize {
```

Comment:
this is not correct, we need to implement the size function by calculating all the fields, we never write to the size variable to read from it